internal/backup_push_handler.go

```diff
@@ -193,6 +193,21 @@ func HandleBackupPush(uploader *Uploader, archiveDirectory string) {
 		tracelog.ErrorLogger.Printf("Failed to upload sentinel file for backup: %s", backupName)
 		tracelog.ErrorLogger.FatalError(err)
 	}
+	// If pushing permanent delta backup, mark all previous backups permanent as well
+	if isPermanent && currentBackupSentinelDto.IsIncremental() {
```

Comment:
There are several thoughts about metadata files:
1. Now metadata contains vital info about backup permanence property so we should fail if we can't upload it.
2. We should upload metadata files in order of backups. It's gonna be a fail if we mark as permanent only last delta backup but don't mark as permanent its base.